storage/rocksdb/ha_rocksdb.cc

```diff
@@ -15229,16 +15238,24 @@ ha_rows ha_rocksdb::multi_range_read_info_const(uint keyno, RANGE_SEQ_IF *seq,
       // Indicate that we will use MultiGet MRR
       *flags &= ~HA_MRR_USE_DEFAULT_IMPL;
       *flags |= HA_MRR_SUPPORT_SORTED;
-      *bufsz = mrr_get_length_per_rec() * res * 1.1 + 1;
+      uint calculated_buf = mrr_get_length_per_rec() * res * 1.1 + 1;
+      // The passed in bufsz contains maximum available buff size --- by
+      // default, its value is specify by session variable read_rnd_buff_size,
+      // final bufsz shouldn't bigger than its limit
+      *bufsz = std::min(*bufsz, calculated_buf);
     }
   } else {
     // For scans on secondary keys, we use MultiGet when we read the PK values.
     // We only need PK values when the scan is non-index-only.
     if (!(*flags & HA_MRR_INDEX_ONLY)) {
-      *flags &= ~HA_MRR_SUPPORT_SORTED; //  Non-sorted mode
+      *flags &= ~HA_MRR_SUPPORT_SORTED;  //  Non-sorted mode
       *flags &= ~HA_MRR_USE_DEFAULT_IMPL;
       *flags |= HA_MRR_CONVERT_REF_TO_RANGE;
-      *bufsz = mrr_get_length_per_rec() * res * 1.1 + 1;
+      uint calculated_buf = mrr_get_length_per_rec() * res * 1.1 + 1;
+      // The passed in bufsz contains maximum available buff size --- by
+      // default, its value is specify by session variable read_rnd_buff_size,
+      // final bufsz shouldn't bigger than its limit
+      *bufsz = std::min(*bufsz, calculated_buf);
```

Comment:
Should this also be capped by `@@mrr_batch_size`? This would allow to use a smaller buffer.

(note: for the BKA, there's no way to control this. See JOIN_CACHE_BKA::aux_buffer_incr(),  where they assume that MRR buffer usage is proportional to #rows read, and it consumes `tab->file->stats.mrr_length_per_rec` per row read)

Reply:
@lth also mentioned this offline. let me update this code to get min of three data